### ISSUE TITLE
storage volume download query requests keep sending after pool deleted

### ISSUE THREAD

**Issue by lvroyce** [kimchi-project/kimchi#624]
When a storage pool contains a downloading volume, it can be deactivated and deleted, but its query task requests are keep sending.
##  Reproduce
1. Build and active a directory based storage pool
1. Download a volume in this storage pool
1. Before downloading finishes, deactivate and delete the storage pool
1. check kimchi log and we can see query progress of downloading is on going.

## Expected result
We don't allow delete a storage pool which contains download task, or we just terminate the task. Or at least we make front end does not query it.

**Comment by cd1** [comment]
The backend keeps downloading the file even after undefining the storage pool, so not querying the server isn't a solution because the file would still be downloaded anyway, and we would waste resources on the server.

We need to block the user (both in the backend and in the frontend)  from deactivating a storage pool with running downloads, or kill the download tasks. Which approach would be the best solution now? Blocking the user from deactivating the pool would be nice if they could cancel the downloads manually but that feature doesn't exist yet; and killing the current downloads seems too aggressive for me (e.g. what if the user prefered to wait for a download to finish if they knew it was still running?).

**Comment by cd1** [comment]
In order to kill existing download tasks, first we need to:

1. Solve issue #78.
2. Design some way to differentiate the storage volume creation tasks; we need to know exactly which of them are downloads. Currently, every volume creation method starts a task with the URI `/storagepools/<pool>/storagevolumes/<volume>`, and none of them indicate whether it's a download, an upload, a regular capacity-based creation...

**Comment by lvroyce** [comment]
I don't think we need to differentiate whether a task is "upload/download". If a storage pool is deleted, all tasks on it need to be stopped no matter what type of task it is. For me the task termination need to be initiated by backend model function, and UI will not take care of task termination. So I think for current at least ,we can make front end stop querying these tasks.

**Comment by cd1** [comment]
You're right, I guess we don't need to differentiate the type of volume creation; we'll need to kill all tasks belonging to that storage pool anyway.
